### PR TITLE
Add two synthetic features: class predictions and class probabilities into classifcation operators

### PR DESCRIPTION
**Pull request by weixuanfu2016**
[please review the [contribution guidelines](http://rhiever.github.io/tpot/contributing/) prior to submitting your pull request. go ahead and delete this line if you've already reviewed said guidelines.]
## What does this PR do?

Add two synthetic features: class predictions and class probabilities into classifcation operators
## Where should the reviewer start?

Line 214 in export_utils.py
## How should this PR be tested?

 MNIST example
## What are the relevant issues?
#214
## Questions:

@teaearlgraycold : The old synthetic feature (only adding class predictions) can work for both classifier and regressor. I kept it for regressor. But I am not sure whether it is reasonable put regressor into VotingClassifier. Will it output predicted (expected) values to next regressor in a pipeline?

### CODE REVIEW

File: tpot/export_utils.py [hunk 1]
```diff
@@ -209,18 +209,24 @@ def process_operator(operator, depth=0):
...
-                format(tpot_op.export(*args))
-            )
+        # add its class probabilities and class prodictions as synthetic features and only for classifier
+        if tpot_op.root  and depth > 0:
+            # only for classifier unless will casue error when using regressor
+            if tpot_op.classification:
+                step = "make_union(make_pipeline(VotingClassifier([(\"est_prob\", {})], voting = \"soft\"), ".format(tpot_op.export(*args))
+                step += "FunctionTransformer(lambda X: X[0], validate=False)), "
+                step += "VotingClassifier([(\"est_class\", {})], voting = \"hard\")".format(tpot_op.export(*args))
+                step += ", FunctionTransformer(lambda X: X))"
```
Comment (teaearlgraycold) [inline tpot/export_utils.py]: Should probably just use a multi-line string here.

```Python
step = ("make_union("
"    make_pipeline("
"        VotingClassifier([("est_prob", {OP_CODE})], voting="soft"),"
"        FunctionTransformer(lambda X: X[0], validate=False)"
"    ),"
"    VotingClassifier([("est_class", {OP_CODE})], voting="hard"),"
"    FunctionTransformer(lambda X: X)"
")").format(OP_CODE=tpot_op.export(*args))
```

Comment (weixuanfu2016) [inline tpot/export_utils.py]: Thanks for your suggestions. Codes are revised based on it.

**Comment by rhiever** [comment]
Please write unit tests that verify the functionality of this new method.

**Comment by weixuanfu2016** [comment]
I add a unit test but don't know why it cause one test failed above. The nose tests is all right in all three of my local devices in three difficult OS (Windows, Linux and MacOS).

**Comment by weixuanfu2016** [comment]
I closed this old PR since the PR #460 covered new function in this PR already

**Pull request closed without merging**